plugins/net_plugin/net_plugin.cpp

```diff
@@ -1686,8 +1710,15 @@ namespace eosio {
       }
    }
 
-   void big_msg_manager::recv_transaction (connection_ptr c) {
+   void big_msg_manager::recv_transaction (connection_ptr c, const transaction_id_type& tid) {
       pending_txn_source = c;
+      if (c &&
```

Comment:
The reset logic contained in this if is repeated twice and should be pulled out into a member function of connection.